src/payment/strategies/checkoutcom-custom/checkoutcom-apm/checkoutcom-apm-payment-strategy.ts

```diff
@@ -0,0 +1,43 @@
+import { InternalCheckoutSelectors } from '../../../../checkout';
+import { OrderRequestBody } from '../../../../order';
+import { PaymentArgumentInvalidError } from '../../../errors';
+import { PaymentInstrument, WithDocumentInstrument } from '../../../payment';
+import { PaymentRequestOptions } from '../../../payment-request-options';
+import CheckoutcomCustomPaymentStrategy from '../checkoutcom-custom-payment-strategy';
+
+export default class CheckoutcomAPMPaymentStrategy extends CheckoutcomCustomPaymentStrategy {
+
+    protected async _executeWithoutHostedForm(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+        const paymentData = payment && payment.paymentData;
+
+        if (!payment || !paymentData) {
```

Comment:
you probably only need to check for paymentData as you are already checking for payment above.

`if (!paymentData)`

Reply:
This one was the only condition that I couldn’t  change because the `payment` object was checked by typescript as possible `undefined` so this check assures that when any method called from `payment` don't will get an error